src/ui/server/controllers/insurance/business/company-details/index.POST-continue.test.ts

```diff
@@ -73,5 +101,18 @@ describe('controllers/insurance/business/companies-details', () => {
         expect(res.redirect).toHaveBeenCalledWith(ROUTES.PROBLEM_WITH_SERVICE);
       });
     });
+
+    describe('when api.keystone.application.update.exporterCompany does not return success', () => {
+      beforeEach(() => {
+        res.locals = { csrfToken: '1234' };
+        api.keystone.application.update.exporterCompany = jest.fn(() => Promise.reject());
+      });
+
+      it(`should redirect to ${ROUTES.PROBLEM_WITH_SERVICE}`, () => {
+        post(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(ROUTES.PROBLEM_WITH_SERVICE);
+      });
+    });
```

Comment:
Can we wrap this in a `describe('api error handling', () => {` 

Also we should be stubbing `mapAndSave.companyDetails` instead of the keystone API because this is what the controller actually calls,

We can then add another test where if the call returns `Promise.resolve(false)`

Check out e.g single contract policy POST controller test